server/container/util-java8/src/test/java/org/apache/james/util/CompletableFutureUtilTest.java

```diff
@@ -95,4 +96,45 @@ public void allOfShouldWorkOnVeryLargeStream() {
                 .map(CompletableFuture::completedFuture))
             .join();
     }
+
+    @Test
+    public void mapOnAllShouldMapOnStreamInsideACompletableFuturOfStream() {
+        CompletableFuture<Stream<Integer>> futurOfInteger = CompletableFuture.completedFuture(Stream.of(1,2,3));
+
+        assertThat(
+            CompletableFutureUtil.map(futurOfInteger, integer ->
+                integer * 2)
+                .join()
+                .collect(Guavate.toImmutableList()))
+            .containsExactly(2,4,6);
+    }
+
+    @Test
+    public void flatMapOnAllShouldMapOnStreamInsideACompletableFuturOfStreamAndTransformTheResultingStreamOfCompletableFutureIntoACompletableOfStreamAndFlapIt() {
+        CompletableFuture<Stream<Integer>> futurOfInteger = CompletableFuture.completedFuture(Stream.of(1,2,3));
+
+        assertThat(
+            CompletableFutureUtil.thenComposeOnAll(futurOfInteger, integer ->
+                CompletableFuture.completedFuture(integer * 2))
+                .join()
+                .collect(Guavate.toImmutableList()))
+            .containsExactly(2,4,6);
+    }
+
+    @Test
+    public void keepValueShouldCompleteWhenTheGivenCompletableFutureEnd() {
```

Comment:
IMO null is not a edge case in this case. Null could be an edge case for other method just like -1, 0, Integer.MAX and Integer.MIN would be for some other method.
But here looking at the code and the type signature, it's really obvious the function is not going to have a different comportment with null than with 42.